src/crunchy_copy.py

```diff
@@ -298,6 +300,31 @@ def process(self):
         signal_dead_mans_snitch(self.cluster["name"])
 
 
+def validate_target(target: Optional[str] = None):
+    """
+    Determine if the backup target is valid.
+
+    This should check to see if the string representation of the date
+    is a date and whether it's a valid Saturday.
+
+    It should return the string version of the date so it can
+    be used in file paths with S3.
+    """
+    if not target:
+        target_date = datetime.now()
+    else:
+        try:
+            target_date = datetime.strptime(target, "%Y%m%d")
+        except ValueError:
+            raise argparse.ArgumentTypeError(
+                f"{target} is not a valid date. It must be in the format YYYYMMDD."
+            )
+
+    if not is_valid_saturday(target_date):
+        raise ValueError(f"Today {target_date} is not a valid Saturday")
```

Comment:
nitpick: Can drop the "Today" from the error string since it might not be today's date that's being processed.